### ISSUE TITLE
Mistake in the description of equality comparison for linked signal

### ISSUE THREAD

**Issue by nickregeda** [angular/angular#59094]
### Describe the problem that you experienced

Please check this section
https://angular.dev/guide/signals/linked-signal#custom-equality-comparison

There is an error in this code because the parameters for the callback in equal are strings, i.e. the email value itself, not the activeUser value
Error: Property id does not exist on type string
```
const activeUser = signal({id: 123, name: 'Morgan'});
const email = linkedSignal(() => `${activeUser().name}@example.com`, {
  // Consider the user as the same if it's the same `id`.
  equal: (a, b) => a.id === b.id,
});
```

### Enter the URL of the topic with the problem

https://angular.dev/guide/signals/linked-signal#custom-equality-comparison

### Describe what you were looking for in the documentation

I'm familiarizing myself with the updates that were introduced in Angular version 19.

### If this problem caused an exception or error, please paste it here

```true

```

### If the problem is browser-specific, please specify the device, OS, browser, and version

```true

```

### Provide any additional information here in as much as detail as you can

```true

```

**Comment by nickregeda** [comment]
`const activeUser = signal({id: 123, name: 'Morgan'});
const email = linkedSignal(() => `${activeUser().name}@example.com`, {
  // Consider the user as the same if it's the same `id`.
  equal: (a, b) => a.id === b.id,
});
// Or, if separating `source` and `computation`
const alternateEmail = linkedSignal({
  source: activeUser,
  computation: user => `${user.name}@example.com`,
  equal: (a, b) => a.id === b.id,
});
// This update to `activeUser` does not cause `email` or `alternateEmail`
// to update because the `id` is the same.
activeUser.set({id: 123, name: 'Morgan', isAdmin: false});`

The whole example in the “Custom equality comparison” section is wrong, because equal takes the values (a, b) of the computation result
And they are the strings in this example

**Comment by saikorthivada** [comment]
@nickregeda It seems the issue persists. The `equals` function is focusing on the returned value from the computation callback rather than the original signal. I tested a different approach, and it worked as expected.

### TS Code
```
import { CommonModule } from '@angular/common';
import { Component, computed, linkedSignal, signal } from '@angular/core';

interface IUser {id: number, name: string}
@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  imports: [CommonModule],
  styleUrl: './app.component.scss'
})
export class AppComponent {
  title = 'v-19';
  activeUser = signal<any>({id: 123, name: 'Morgan'});
  email = linkedSignal(() => ({email:`${this.activeUser().name}@example.com`, ...this.activeUser()}), {
    equal: (a, b) => a.id == b.id,
  });

  alternateEmail = linkedSignal({
    source: this.activeUser,
    computation: user => ({email:`${user.name}@example.com`, ...user}),
    equal: (a, b) => a.id === b.id,
  });

  constructor() {
    setTimeout(() => {
    this.activeUser.set({id: 1234, name: 'Morgan123', isAmin: false});
    }, 3000);
  }
}
```
### HTML Code
```

<p>{{alternateEmail().email}}</p>
<p>{{email().email}}</p>
```

In the example above, instead of returning a string value from the computation, I returned a full object with the required property added. This resolved the issue, and it started working as expected.

It seems the behavior described in the documentation differs from how it actually works.

Uncertain About the Expected Behavior of This Feature

**Comment by nickregeda** [comment]
@saikorthivada 
That's right
equal is specified for a particular signal and defines the condition by which it will be evaluated as updated or not, and thus whether or not it will trigger computed or linkedSignal or effect.
That doesn't mean we can just specify equal for the source signal in linkedSignal and hope that it will use its values in params, that's it

**Comment by saikorthivada** [comment]
@nickregeda  yeah you are right. Need to know how exactly this should work from Angular team. Great finding.

**Comment by JeanMeche** [comment]
The equal function is used to determine if the signal itself was changed. The dirtyness of the source signal is determined by that said signal. 
The example should probably be 

```ts
const activeUser = signal({id: 123, name: 'Morgan', isAdmin: true});
const email = linkedSignal(() => ({id:`${activeUser().name}@example.com`}), {
  // Consider the user as the same if it's the same `id`.
  equal: (a, b) => a.id === b.id,
});
// Or, if separating `source` and `computation`
const alternateEmail = linkedSignal({
  source: activeUser,
  computation: user => ({id:`${user.name}@example.com`}),
  equal: (a, b) => a.id === b.id,
});
// This update to `activeUser` does not cause `email` or `alternateEmail`
// to update because the `id` is the same.
activeUser.set({id: 123, name: 'Morgan', isAdmin: false});
```

**Comment by bst-celum** [comment]
The documentation still feels a bit misleading to me and might cause some confusion.

The original idea seems to be that the `email` signal should only update when the `id` of the `activeUser` changes. However, in the current example, the `email` signal produces an object with an `id` property, which feels out of place.

Here are the main issues I see:

Type Mismatch: The `email` signal is producing an object `{ id: ... }` instead of a `string` (the actual email). This doesn't align with what you'd expect given the name and context.
Confusion Around Equality Check: The equality function is comparing the `id` property of the result of the computation, but the description implies that it’s tied to changes in the `activeUser` signal (the source). This creates a disconnect between the intended behavior and what’s actually happening in the code.

**Comment by pkozlowski-opensource** [comment]
@bst-celum you are completely right, the code example is misleading. 

I've sent https://github.com/angular/angular/pull/59532 to propose _something_ but open to other suggestions.
